extensions/mssql/src/features.ts

```diff
@@ -31,46 +32,77 @@ export class TelemetryFeature implements StaticFeature {
-			if (!tenant) {
-				window.showErrorMessage(unauthorizedMessage);
-				return undefined;
-			}
-			let tokenBundle = securityToken[tenant.id];
-			if (!tokenBundle) {
-				window.showErrorMessage(unauthorizedMessage);
+	protected async getToken(request: contracts.RequestSecurityTokenParams): Promise<contracts.RequestSecurityTokenResponse | undefined> {
+		const accountList = await azdata.accounts.getAllAccounts();
+		let account: azdata.Account;
+
+		if (accountList.length < 1) {
+			// TODO: Prompt user to add account
+			window.showErrorMessage(localize('mssql.missingLinkedAzureAccount', "Azure Data Studio needs to contact Azure Key Vault to access a column master key for Always Encrypted, but no linked Azure account is available. Please add a linked Azure account and retry the query."));
+			return undefined;
+		} else if (accountList.length > 1) {
+			let options: QuickPickOptions = {
+				ignoreFocusOut: true,
+				placeHolder: localize('mssql.chooseLinkedAzureAccount', "Please select a linked Azure account:")
+			};
+			let items = accountList.map(a => new AccountFeature.AccountQuickPickItem(a));
+			let selectedItem = await window.showQuickPick(items, options);
+			if (!selectedItem) { // The user canceled the selection.
+				window.showErrorMessage(localize('mssql.canceledLinkedAzureAccountSelection', "Azure Data Studio needs to contact Azure Key Vault to access a column master key for Always Encrypted, but no linked Azure account was selected. Please retry the query and select a linked Azure account when prompted."));
 				return undefined;
 			}
-
-			let params: contracts.RequestSecurityTokenResponse = {
-				accountKey: JSON.stringify(account.key),
-				token: securityToken[tenant.id].token
-			};
-
-			return params;
-		});
+			account = selectedItem.account;
+		} else {
+			account = accountList[0];
+		}
+
+		const securityToken: { [key: string]: any } = await azdata.accounts.getSecurityToken(account, azdata.AzureResource.AzureKeyVault);
+		const tenant = account.properties.tenants.find((t: { [key: string]: string }) => request.authority.includes(t.id));
+		const unauthorizedMessage = localize('mssql.insufficientlyPrivelagedAzureAccount', "The configured Azure account for {0} does not have sufficient permissions for Azure Key Vault to access a column master key for Always Encrypted.", account.key.accountId);
+		if (!tenant) {
+			window.showErrorMessage(unauthorizedMessage);
+			return undefined;
+		}
+		let tokenBundle = securityToken[tenant.id];
+		if (!tokenBundle) {
+			window.showErrorMessage(unauthorizedMessage);
+			return undefined;
+		}
+
+		let params: contracts.RequestSecurityTokenResponse = {
+			accountKey: JSON.stringify(account.key),
+			token: securityToken[tenant.id].token
+		};
+
+		return params;
 	}
+
+	static AccountQuickPickItem = class implements QuickPickItem {
```

Comment:
Any particular reason to have this defined here instead of as a local class at the file level?

Reply:
The scope of this `QuickPickItem` implementation is local to this enclosing `AccountFeature` so I added it here.

Reply:
I'm not sure if making it static here has any significance but you shouldnt need static

Reply:
Without static, I was getting the error:
`Property 'AccountQuickPickItem' does not exist on type 'typeof AccountFeature'`

Reply:
I assume when you did that you didn't remove the AccountFeature part on line 62 which is why it gave that error.

That being said there's no reason this shouldn't be static since having a separate definition for each instance doesn't make sense.

Reply:
Oh yea. 😆 In all the chaos I overlooked such a simple thing.
Fixed.

Reply:
Well, I don't think that's correct though. Having it be static makes sense in this case.

Reply:
Okay, reverted.